cpp/src/arrow/python/flight.cc

```diff
@@ -196,9 +200,14 @@ PyGeneratorFlightDataStream::PyGeneratorFlightDataStream(
   generator_.reset(generator);
 }
 
-std::shared_ptr<arrow::Schema> PyGeneratorFlightDataStream::schema() { return schema_; }
+std::shared_ptr<Schema> PyGeneratorFlightDataStream::schema() { return schema_; }
+
+Status PyGeneratorFlightDataStream::GetSchemaPayload(FlightPayload* payload) {
+  return ipc::internal::GetSchemaPayload(*schema_, &dictionary_memo_,
```

Comment:
I'm not sure I understand. This seems to populate a `DictionaryMemo` but it's not used afterwards?

Reply:
We don't have a test for dictionary arrays in `test_flight.py`...

Reply:
Last I tried, cross-language Flight with dictionaries still didn't work, or even from C++ to C++, so it wouldn't have worked before in Python. https://issues.apache.org/jira/browse/ARROW-5143

Reply:
Ah... They work with `DoGet` but perhaps not with `DoPut` then?

Reply:
@lihalite I'm fixing dict transfer with DoPut as part of ARROW-5113. It may produce conflicts for both you and @wesm  :-)

Reply:
OK, you may want to abort the dict transfer work until this is merged